src/main/java/Parser.java

```diff
@@ -0,0 +1,139 @@
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+
+public class Parser {
+    private BufferedReader br;
+    private TaskList listOfTasks;
+    private Storage storage;
+    private UI ui;
+
+    public Parser(TaskList tasks, Storage storage, UI ui) {
+        this.br = new BufferedReader(new InputStreamReader(System.in));
+        this.listOfTasks = tasks;
+        this.storage = storage;
+        this.ui = ui;
+    }
+
+    public void parse() throws Exception, DukeException {
```

Comment:
Might it be possible to segregate your method to smaller methods? The method will works fine without it but i believe that overall it will looks neater and more concise.

Reply:
Dear Jun Hong T,

There is definitely a better way of handing the method, I will do what I can to segregate this specific method and making it more readable. Great point to make Jun Hong T, amazing comment!

Best regards
Roby Tanama